help_to_heat/frontdoor/interface.py

```diff
@@ -71,6 +77,47 @@ class SuccessSchema(marshmallow.Schema):
     success = marshmallow.fields.Boolean()
 
 
+def get_url(postcode: str, offset: int, max_results: int) -> str:
+    url = f"https://api.os.uk/search/places/v1/postcode?maxresults={max_results}&postcode={postcode}&lr=EN&dataset=DPA,LPI&key={settings.OS_API_KEY}"
+    if (offset):
+        url += f"&offset={offset}"
+    return url
+
+def get_addresses_from_api(postcode) -> List[Dict]:
+    try:
+        max_results_number = 100
+        offset = 0
+
+        url = get_url(postcode=postcode, offset=offset, max_results=max_results_number)
+        response = requests.get(url)
+        response.raise_for_status()
+        json_response = response.json()
+
+        total_results_number = json_response['header']["totalresults"]
+        api_results = json_response['results']
+        api_results_all = api_results
+
+        if total_results_number > max_results_number:
```

Comment:
maybe it's Monday morning brain but I'm finding this a little tricky to follow, can we put it in a function so we can describe what it's doing?

Reply:
Agree it's a bit hard to follow.. how about adding a comment? I tried pulling out a function but that didn't look good :( I tried refactoring the `get_url` function to `get_json_response_from_url`, and it looks slightly better?

Reply:
Comment is acceptable instead of pulling out a function :)

Reply:
Cool, added some explanation and kept the latest refactoring :)